code/modules/events/event_dynamic.dm

```diff
@@ -45,45 +45,128 @@ var/list/event_last_fired = list()
+	possibleEvents[/datum/event/spider_infestation] = 50 + 25 * active_with_role["Security"]
+	possibleEvents[/datum/event/carp_migration] = 50 + 25 * active_with_role["Security"]
+
+	possibleEvents[/datum/event/communications_blackout] = 40//Comms blackout is too common and annoying
+
+	//Ion laws are fun!
+	possibleEvents[/datum/event/ionstorm] = active_with_role["AI"] * 35 + active_with_role["Cyborg"] * 20 + active_with_role["Engineer"] * 5 + active_with_role["Scientist"] * 5
+
+	possibleEvents[/datum/event/grid_check] = 100
+
+	//Janitor has a big weight because they usually fix the lights
+	possibleEvents[/datum/event/electrical_storm] = 60 + 20 * active_with_role["Janitor"] + 5 * active_with_role["Engineer"]
+
+	//Rad storms no longer medical-dependant. If there's no medical you can drink tea and vodka
+	possibleEvents[/datum/event/radiation_storm] = 100
+
+
+	possibleEvents[/datum/event/meteor_shower] = 50 + 25 * active_with_role["Engineer"]//Quite dependant on engineers
 
 	if(active_with_role["Medical"] > 0)
-		possibleEvents[/datum/event/radiation_storm] = active_with_role["Medical"] * 10
-		possibleEvents[/datum/event/spontaneous_appendicitis] = active_with_role["Medical"] * 10
+		//These two are just pointless without medical, so no base chance
+		possibleEvents[/datum/event/spontaneous_appendicitis] = active_with_role["Medical"] * 25
 		possibleEvents[/datum/event/viral_infection] = active_with_role["Medical"] * 10
 
-	possibleEvents[/datum/event/prison_break] = active_with_role["Security"] * 50
-	if(active_with_role["Security"] > 0)
-		if(!sent_spiders_to_station)
-			possibleEvents[/datum/event/spider_infestation] = max(active_with_role["Security"], 5) + 5
-		possibleEvents[/datum/event/random_antag] = max(active_with_role["Security"], 5) + 2.5
+	if (minutes_passed > 60)
+		//No point in a prison break when no antags have been arrested yet
+		possibleEvents[/datum/event/prison_break] = active_with_role["Security"] * 15 + active_with_role["Cyborg"] * 20
+
+		//Give the previous antag a chance to tell their story before spawning another
+		//This is generally less common but scales with playercount
+		possibleEvents[/datum/event/random_antag] = min(30, player_list.len) + active_with_role["Security"]*8
+
+
+
+
+
+
+
+
+
+
+
+
+	//Severe Events
+	//=================================================
+	//Note, carp migration and viral infection have different behaviour depending whether they're called as moderate or severe
+	//So they exist in both categories
+	possibleEvents[/datum/event/carp_migration] = 60 + 10 * active_with_role["Security"]
+	possibleEvents[/datum/event/viral_infection] = active_with_role["Medical"] * 25
+
+	//More heavily weighted towards engineering, meteor storms are punishing without shields
+	possibleEvents[/datum/event/meteor_wave] = 50 + 15 * active_with_role["Engineer"]
+	possibleEvents[/datum/event/blob] = 40 + 5 * active_with_role["Engineer"] + 5 * active_with_role["Security"]
+
```

Comment:
Aaand skull beat me to it.

The delay to get the engine setup did nothing, because as he already said, severe events are delayed for something like 60-90 minutes anyway. So just removing obsolete code